replica/table.cc

```diff
@@ -66,6 +66,33 @@ static seastar::metrics::label keyspace_label("ks");
 
 using namespace std::chrono_literals;
 
+sstables::generation_type table::make_new_generation(std::optional<sstables::generation_type> prev) {
+    auto prev_value = prev.value_or(sstables::generation_type(0)).value();
```

Comment:
what about we merge calculate_generation_for_new_table() into this function (and then get rid of the former)? 

so we do:
auto prev_value = prev.value_or(_sstable_generation).value();

as we don't want to ignore the known generation

Also please rename make_new_generation to make_new_sstable_generation.